src/Extractor/SourceTargetMappingExtractor.php

```diff
@@ -17,6 +17,18 @@
  */
 class SourceTargetMappingExtractor extends MappingExtractor
 {
+    public function getProperties(string $class): iterable
+    {
+        return array_values(
+            array_unique(
+                [
+                    ...parent::getProperties($class),
+                    ...$this->getConstructorParameters($class),
```

Comment:
It will fetch properties from the constructor in the source class also which is not wanted in some case

Can we change the signature of this method like : 

```
public getProperties(string $class, bool $withConstructorParamaters = false)
```

Or have two differents methods (one for source, one for target)

And then we would only use it when wanted ? 

(Then we would keep the implementation in the abstract class only)